js/text-area/textarea-autoresize.ts

```diff
@@ -19,9 +19,41 @@ function onDelegatedEvent(
 }
 
 function update_height(target: HTMLTextAreaElement) {
-  // Automatically resize the textarea to fit its content.
-  target.style.height = "auto";
-  target.style.height = target.scrollHeight + "px";
+  if (target.scrollHeight > 0) {
+    // Automatically resize the textarea to fit its content.
+    target.style.height = "auto";
+    target.style.height = target.scrollHeight + "px";
+  } else {
+    // The textarea is not visible on the page, therefore it has a 0 scroll height.
+
+    // If `.getAttribute("rows")` is set to 0 (as `target.rows` is always > 0 and
+    // invalid values use the default value of `2`), then we can wait for the textarea
+    // to become visible and call `update_height` again as the scroll height should not
+    // be 0.
+    //
+    // Note: `rows` is only set to `0` from the python code when the textarea is
+    // autoresized and no `rows` value is given.
+    const targetRows = target.getAttribute("rows");
+    if (targetRows == "0") {
```

Comment:
I can't seem to follow why it's beneficial to use `rows=0` as an additional signal to resize (when visible) instead of always updating the height?

Also, wouldn't this mean that if a user specifies `rows=n`, then you don't get the same resize-on-visible behavior?

Reply:
You would not get it.  But it would resize.

I'm thinking we need to add an extra attribute to avoid the confusion.

Reply:
Yes. Removed.

It should always be put into the intersection observer if scrollHeight is 0. Removing the `rows=0` init value change.